test/telemetry_metrics_statsd_test.exs

```diff
@@ -317,6 +317,7 @@ defmodule TelemetryMetricsStatsdTest do
       udp = TelemetryMetricsStatsd.get_udp(reporter)
 
       TelemetryMetricsStatsd.udp_error(reporter, udp, :closed)
+      Process.sleep(100)
```

Comment:
Can you add the `liveness` library to the test dependencies and use it to assert the eventually we get a new socket? https://hexdocs.pm/liveness/readme.html#use-in-tests

Reply:
done